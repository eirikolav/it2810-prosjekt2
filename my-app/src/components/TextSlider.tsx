import React from "react"
import './Frame.css'
/*import { stringify } from "querystring";*/
import './Slider.css'


interface TextSliderProps {
    textHandleClick: () => void;
    index: number;

}

interface TextSliderState {
    text: string[];

}

export default class textSlider extends React.Component<any, TextSliderProps, TextSliderState> {
    poems = [
        "https://poetrydb.org/title/Not at Home to Callers/.json",
        "https://poetrydb.org/title/The words the happy say/.json",
        "https://poetrydb.org/title/The Life we have is very great./.json"
    ]

    data = [] as string[][];


    componentDidMount() {
        var i;
        for (i = 0; i < this.poems.length; i++) {
            fetch(this.poems[i])
                .then((response) => response.json())
                .then((result) => {
                    this.data.push(result[0].lines)
                });
        }
    };

    makeText(index: number) {
        if (this.data != null) {
            return (this.data.map((list: string[], index: number) => [list.map((line: string, index: number) => <p key={index}>{line}</p>)])[index]);
        }
        else {
            return <p> ... loading ...</p>
        }
    }


    render() {
        return (
            <div>
                <div className="wrap">
                    <div className="poem"> {this.makeText(this.props.index)} </div>
                </div>
                <div>
                    <button onClick={this.props.textHandleClick} className="textbutton ok" > Tekst </button>
                </div>


            </div>
        )
    }
}






