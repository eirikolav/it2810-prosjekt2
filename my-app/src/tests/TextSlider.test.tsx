import React from 'react'
import renderer from 'react-test-renderer'
import TextSlider from "../components/TextSlider"

test("Test Header", () => {
    const tree = renderer.create(<TextSlider />).toJSON()
    expect(tree).toMatchSnapshot()
})