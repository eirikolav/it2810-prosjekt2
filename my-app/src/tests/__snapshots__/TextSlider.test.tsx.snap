// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header 1`] = `
<div>
  <button>
     Tekst 
  </button>
</div>
`;
